### DISCUSSION TITLE
Difficulty (strain) graph

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#30116] [Ideas]
I want to add (implement) difficulty graph as an UI element in the game.
So I would like to hear design propositions.

cpol is using 2 color system (giving more information), most likely dividing strain into aim and speed, but lazer solution obviously should be more generic. 
Some other graphs are using one graph, what can be quite problematic.

The reason why it's problematic is because there so single "strain" graph in the pp. Most graphs solve this by summing strain values from different skills. But this can be very misleading when aim and speed values are the same (the most notable example is Save Me having very big spike in the end, what can be quite misleading)

My solution to this is 2 colored graph:
- first is for the biggest strain out of all skills
- second is summed strain (not exactly as in pp, but with bigger power, as single-strain graphs already do)

Those are references
cpol:
![image](https://github.com/user-attachments/assets/052284d0-fe71-4f7e-82d0-e80b47b9638e)

BTMC overlay:
![image](https://github.com/user-attachments/assets/dcf46d45-a347-471e-a373-57a7a20a80e4)

Save me spiking problem
Old cpol:
![image](https://github.com/user-attachments/assets/c07a9bcc-cf66-45de-9c46-540ba5c0a404)

Bathbot:
![image](https://github.com/user-attachments/assets/c869eb56-5075-418e-a9ac-577a31d61c57)

**Comment by peppy** [comment]
This would be something that would either replace or be a toggle for the "object density"  on the song progress display. It would also be fine to use it in a custom component with more fidelity like the above proposals.

I'm not against "peak" and "rolling average" lines on the graph, both of these are important to have.
